core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -589,6 +658,16 @@ private void verifyGroupConfig(Admin client, String groupName, Map<String, Strin
         }, 10000, config + " are not updated");
     }
 
+    private void verifyClientMetricsConfig(Admin client, String clientMetricsName, Map<String, String> config) throws Exception {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.CLIENT_METRICS, clientMetricsName);
+        TestUtils.waitForCondition(() -> {
+            Map<String, String> current = getConfigEntryStream(client, configResource)
+                    .filter(configEntry -> Objects.nonNull(configEntry.value()))
+                    .collect(Collectors.toMap(ConfigEntry::name, ConfigEntry::value));
+            return config.entrySet().stream().allMatch(e -> e.getValue().equals(current.get(e.getKey())));
```

Comment:
`allMatch` is always `true` when the `config` is empty. Maybe we should check the `current.isEmpty` instead.

Reply:
I don't quite understand. It is true that `allMatch` is "vacuously" true when stream is empty, but in some cases the `config` parameter is not empty.

Reply:
Sorry for unclear comments.

My point was the test case which want to "delete all configs" should check `current.isEmpty()` instead. For example:
```java
    private void verifyClientMetricsConfig(Admin client, String clientMetricsName, Map<String, String> config) throws Exception {
        ConfigResource configResource = new ConfigResource(ConfigResource.Type.CLIENT_METRICS, clientMetricsName);
        TestUtils.waitForCondition(() -> {
            Map<String, String> current = getConfigEntryStream(client, configResource)
                    .filter(configEntry -> Objects.nonNull(configEntry.value()))
                    .collect(Collectors.toMap(ConfigEntry::name, ConfigEntry::value));
            // make sure there are no configs anymore 
            if (config.isEmpty()) return current.isEmpty();
            return config.entrySet().stream().allMatch(e -> e.getValue().equals(current.get(e.getKey())));
        }, 10000, config + " are not updated");
    }
```